metaflow/metaflow_config.py

```diff
@@ -187,13 +187,29 @@ def get_version(pkg):
 
 # PINNED_CONDA_LIBS are the libraries that metaflow depends on for execution
 # and are needed within a conda environment
-def get_pinned_conda_libs():
-    return {
-        'click': '7.0',
-        'requests': '2.22.0',
-        'boto3': '1.9.235',
-        'coverage': '4.5.3'
-    }
+def get_pinned_conda_libs(python_version):
+    if python_version.startswith(("2.7", "3.6", "3.7", "3.8")):
+        return {
+            'click': '7.1.2',
+            'requests': '2.24.0',
+            'boto3': '1.14.47',
+            'coverage': '4.5.4'
+        }
+    elif python_version.startswith("3.5"):
+        return {
+            'click': '7.1.2',
+            'requests': '2.24.0',
+            'boto3': '1.9.88',
+            'coverage': '4.5.1'
+        }
+    elif python_version in ["2", "3"]:
+        raise MetaflowException(
+            "Please specify complete python version number, " +
+            "for example 2.7, 3.7, 3.7.8")
+    else:
+        raise MetaflowException(
```

Comment:
@jasonge27 How about [Python 3.9](https://docs.python.org/3.9/whatsnew/3.9.html)?

Reply:
@savingoyal Python 3.9 has not yet landed in conda-forge or conda main channel. We can ignore it for now?

Reply:
Yes, okay to ignore but Metaflow will throw an error whenever `3.9` makes its way to conda and I am not sure we want that. Given that only `3.5` has different dependencies compared to other versions, how about a simple clause - 
```
if py_version.startswith('3.5'):
    ...
else:
    ...
```

Reply:
@savingoyal good idea! updated the PR according your suggestion